bloom-housing/bloom
# fix: jurisdiction permission array approach

# Pull Request Template

## Issue Overview

This PR addresses #3639 & https://github.com/metrotranscom/doorway/issues/326


## Description

This PR updates the ways listings approval permissions are determined (from env to db). Now, each jurisdiction has a listingsApprovalPermissions filed which stores an array. A null or empty array indicates the field is off and if users are listed, they are the approving roles in the listings approval flow. This allows for easier management of multiple jurisdictions having unique needs. The majority of the lines changes here, however, are related to backend testing and ensuring the functionality of the email services connected to this feature. 
 
## How Can This Be Tested/Reviewed?

Ensure that the previous functionality of the listings approval flow works and specifically, ensure that a jurisdictional admin can create a pending review application and can see different experiences based on the jurisdiction that they are assigned too. 

Note: there is one future consideration in that if a partner can create listings, we'd have to tie them to a jurisdiction in some way.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
